source/core/replica-set-high-availability.txt

```diff
@@ -15,29 +19,226 @@ Replica Set High Availability
    :depth: 1
    :class: singlecol
 
-:term:`Replica sets <replica set>` provide high availability using
-automatic :term:`failover`. Failover allows a :term:`secondary` member
-to become :term:`primary` if the current primary becomes unavailable.
+Replica sets support high availability by automatically calling 
+for elections to select the :term:`primary` member for the cluster. 
+Replica sets can trigger an election in response to a variety of events,
+such as:
+
+- Adding a new node to the replica set,
+- :method:`initiating a replica set <rs.initiate()>`,
+- performing replica set maintenance using methods such as :method:`rs.stepDown()` or :method:`rs.reconfig()`, and
+- the :term:`secondary` members losing connectivity to the primary for more than the configured :rsconf:`timeout <settings.electionTimeoutMillis>` (10 seconds by default).
+
+In the following diagram, the primary node was unavailable for longer
+than the :rsconf:`configured timeout <settings.electionTimeoutMillis>`
+and triggers the :ref:`automatic failover <replication-auto-failover>`
+process. One of the remaining secondaries calls for an election to
+select a new primary and automatically resume normal operations.
+
+.. include:: /images/replica-set-trigger-election.rst
+
+The replica set cannot process write operations until the
+election completes successfully. The replica set can continue to serve 
+read queries if such queries are configured to 
+:ref:`run on secondaries <replica-set-read-preference>`.
+
+.. include:: /includes/fact-retryable-writes-failover-election.rst
+
+The median time before a cluster elects a new primary should not
+typically exceed 11.5 seconds, assuming default :rsconf:`replica
+configuration settings <settings>`. This includes time required to
+mark the primary as :ref:`unavailable <replication-auto-failover>` and
```

Comment:
Maybe add a sentence along the lines of "the former can be affected by changing the electionTimeoutMillis replica set config option"?

Reply:
Done